driver/modern_bpf/definitions/missing_definitions.h

```diff
@@ -0,0 +1,1223 @@
+#ifndef __MISSING_DEFINITIONS_H__
+#define __MISSING_DEFINITIONS_H__
+
+/* This header should include different definitions according to different architectures.*/
+
+/* PF_KTHREAD flag to mark the kernel threads */
+#define PF_KTHREAD 0x00200000
+
+/*=============================== ARCH SPECIFIC ===========================*/
+
+/* 32bit syscall active (only in ARCH_x86) */
+#define TS_COMPAT 0x0002
+
+// arch/arm64/include/asm/thread_info.h
+#define _TIF_32BIT (1 << 22)
```

Comment:
This is a good point, in the first version I had the `#ifdef` here, but I thought that probably the `_TIF_32BIT` will be never used on `x86` and vice versa. To simplify the code I removed it but we can reintroduce it if you think it is necessary

Reply:
I'm not sure if it is 100% necessary, but if a macro/define/function/whatever is meant for a specific use case, my instinct is to prevent people from misusing it as much as possible. I have trust issues, I know 🤷🏻‍♂️

Reply:
Make sense, thank you @Molter73!